rs/cli/src/store.rs

```diff
@@ -0,0 +1,273 @@
+                    }
+                    // Running offline mode and ic-admin version is present.
+                    (true, version_from_file) => {
+                        warn!("Offline mode specified! Will use cached ic-admin version: {}", version_from_file);
+                        version_from_file
+                    }
+                    // Running online mode
+                    //
+                    // There is a cached version of ic-admin
+                    // and the cached version is still younger
+                    // than `DURATION_BETWEEN_CHECKS_FOR_NEW_IC_ADMIN`
+                    (false, version_from_file) if !version_from_file.is_empty() && elapsed <= DURATION_BETWEEN_CHECKS_FOR_NEW_IC_ADMIN => {
+                        info!("Using cached ic admin version: {}", version_from_file);
+                        version_from_file
+                    }
+                    // Either there isn't a cached version at all
+                    // or the `DURATION_BETWEEN_CHECKS_FOR_NEW_IC_ADMIN`
+                    // has passed which means that the cache is invalid.
+                    // Check should be performed
+                    (false, _) => {
+                        info!("Checking for new ic-admin version");
+                        let govn_canister_version = governance_canister_version(network.get_nns_urls()).await?;
+                        debug!(
+                            "Using ic-admin matching the version of governance canister, version: {}",
+                            govn_canister_version.stringified_hash
+                        );
+                        let version = match govn_canister_version.stringified_hash.as_str() {
+                            // This usually happens on testnets deployed
+                            // from the HEAD of branch
+                            "0000000000000000000000000000000000000000" => FALLBACK_IC_ADMIN_VERSION,
+                            v => v,
+                        };
+                        version.to_string()
+                    }
+                };
+
+                let ic_admin = self
+                    .init_ic_admin(&version, network, proceed_without_confirmation, neuron, dry_run)
+                    .await?;
+
+                // Only update file when the sync
+                // with governance has been performed
+                std::fs::write(self.ic_admin_status_file()?, version)?;
+                Ok(ic_admin)
+            }
+        }
+    }
+
+    fn cordoned_features_file(&self) -> anyhow::Result<PathBuf> {
+        let file = self.path().join("cordoned_features.yaml");
+
+        if !file.exists() {
+            info!("Cordoned features file was missing. Creating on path `{}`...", file.display());
+            if let Err(e) = std::fs::write(&file, "") {
+                warn!("Failed to create cordoned features file: {:?}", e);
+                warn!("This is not critical now. If github is offline then it could be");
```

Comment:
```suggestion
                warn!("This is not a critical error, but would be if GitHub is unreachable.");
```

Reply:
This whole log doesn't make sense because that code wouldn't be reached if github is down. I will write a clearer message.